java/code/src/com/redhat/rhn/manager/contentmgmt/ContentManager.java

```diff
@@ -682,6 +682,22 @@ private static void alignEnvironment(ContentEnvironment env, Channel baseChannel
                 .collect(toList());
     }
 
+    private static SoftwareEnvironmentTarget fixTargetProperties(SoftwareEnvironmentTarget tgt, Channel newSource,
+            Channel newParent) {
+        // target already exists -> make sure its parent is set correctly
+        tgt.getChannel().setParentChannel(newParent);
+        // and it has original-clone relation with the source
+        tgt.getChannel().asCloned().ifPresentOrElse(
+                c -> c.setOriginal(newSource),
+                () -> {
+                    log.warn("Channel is not a clone: " + tgt.getChannel() + ". Adding clone info.");
+                    ChannelManager.addCloneInfo(newSource.getId(), tgt.getChannel().getId());
+                    tgt.setChannel((Channel) HibernateFactory.reload(tgt.getChannel()));
+                }
+        );
```

Comment:
For me it seems that this is able to "fix" all wrong clone entries when called.
So I wonder if we need the schema migration at all? 

I do not like the idea to remove also "correct" values and wait until customers build and promoted again.
If possible I would keep also incorrect "cloned from" values if a build / promote later just fix it.

Reply:
The problem happened in my old setup where I had a Content Project in such state, that one channel in content project had 2 clones in the same project (this could happen with the old code when you insert a new environment in the middle of an environment path - new channels are cloned, but the ones in the existing successor environment aren't adjusted). In this case the query I wrote returns 2 rows instead of 1.

I also don't like the fact that the migration wipes all clone information and I have an alternative solution in mind that relies on channel label conventions. I'll present it as an alternative later today, maybe this will be the lesser of 2 evils.

Reply:
I simplified the PR. In the end, the original-clone relation is fixed by building/promoting the project. If the user has an inconsistent project, it'll become consistent eventually.
I also made the code robust enough, that it should handle clone inconsistencies well (e.g. if a channel has more than 1 clone in the project).